packages/wrangler/src/cloudchamber/build.ts

```diff
@@ -0,0 +1,225 @@
+			throw err;
+		});
+		child.stdin.write(credentials.password);
+		child.stdin.end();
+		await new Promise((resolve) => {
+			child.on("close", resolve);
+		});
+	});
+}
+
+export async function constructBuildCommand(options: {
+	imageTag?: string;
+	pathToDocker?: string;
+	pathToDockerfile?: string;
+	platform?: string;
+	customBuildCommand?: string;
+}) {
+	if (
+		typeof options.customBuildCommand !== "undefined" &&
+		options.customBuildCommand !== ""
+	) {
+		return options.customBuildCommand;
+	}
+	// require a tag if we provide dockerfile
+	if (
+		typeof options.pathToDockerfile !== "undefined" &&
+		options.pathToDockerfile !== "" &&
+		(typeof options.imageTag === "undefined" || options.imageTag === "")
+	) {
+		throw new Error("must provide an image tag if providing a docker file");
+	}
+	const dockerFilePath = options.pathToDockerfile;
+	const dockerPath = options.pathToDocker ?? "docker";
+	const imageTag = domain + "/" + options.imageTag;
+	const platform = options.platform ? options.platform : "linux/amd64";
+	const defaultBuildCommand = [
+		dockerPath,
+		"build",
+		"-t",
+		imageTag,
+		"--platform",
+		platform,
+		dockerFilePath,
+	].join(" ");
+
+	return defaultBuildCommand;
+}
+
+// Function for building
+export async function dockerBuild(options: { buildCmd: string }) {
+	const buildCmd = options.buildCmd.split(" ").slice(1);
+	const buildExec = options.buildCmd.split(" ").shift();
+	const child = spawn(String(buildExec), buildCmd, { stdio: "inherit" }).on(
+		"error",
+		(err) => {
+			throw err;
+		}
+	);
+	await new Promise((resolve) => {
+		child.on("close", resolve);
+	});
+}
+
+async function tagImage(original: string, newTag: string, dockerPath: string) {
+	const child = spawn(dockerPath, ["tag", original, newTag]).on(
+		"error",
+		(err) => {
+			throw err;
+		}
+	);
+	await new Promise((resolve) => {
+		child.on("close", resolve);
+	});
+}
+
+export async function push(options: {
+	imageTag?: string;
+	pathToDocker?: string;
+}) {
+	if (typeof options.imageTag === "undefined") {
+		throw new Error("Must provide an image tag when pushing");
+	}
+	// TODO: handle non-managed registry?
+	const imageTag = domain + "/" + options.imageTag;
+	const dockerPath = options.pathToDocker ?? "docker";
+	await tagImage(options.imageTag, imageTag, dockerPath);
+	const child = spawn(dockerPath, ["image", "push", imageTag], {
+		stdio: "inherit",
+	}).on("error", (err) => {
+		throw err;
+	});
+	await new Promise((resolve) => {
+		child.on("close", resolve);
+	});
+}
+
+export function buildYargs(yargs: CommonYargsArgvJSON) {
+	return yargs
+		.positional("PATH", {
+			type: "string",
+			describe: "path for the directory containing the dockerfile",
+			demandOption: true,
+		})
+		.option("tag", {
+			alias: "t",
+			type: "string",
+			demandOption: true,
+			describe: "Tag to use for the built image",
+		})
+		.option("path-to-docker", {
+			type: "string",
+			default: "docker",
+			describe: "path to docker binary",
```

Comment:
The option is called "path-to-docker" but accepts a command (i.e. `"docker"`) so maybe `docker-cmd` is a better name? Path implies a full path.